rshim.c

```diff
@@ -2765,6 +2610,139 @@ static const struct cuse_methods rshim_misc_fops = {
 };
 #endif
 
+/* Rshim file operations routines */
+
+/* ioctl message header. */
+typedef struct {
+  uint32_t addr;
+  uint64_t data;
+} __attribute__((packed)) rshim_ioctl_msg;
+
+enum {
+  RSHIM_IOC_READ = _IOWR('R', 0, rshim_ioctl_msg),
+  RSHIM_IOC_WRITE = _IOWR('R', 1, rshim_ioctl_msg),
+};
+
```

Comment:
Should probably put this into an own header file: rshim_ioctl.h

Reply:
How about leave it to a seperate change later? We probably need to clean up this part a little bit, such as using rshim_fs.c / rshim_fs.h, etc.

Reply:
No problem.